packages/zapp/console/src/components/flytegraph/ReactFlow/transformDAGToReactFlowV2.tsx

```diff
@@ -78,33 +78,33 @@ export const buildReactFlowDataProps = (props: BuildDataProps) => {
   const nodeExecutionStatus = mapNodeExecutionStatus();
 
   const cacheStatus: CatalogCacheStatus =
-    nodeExecutionsById[node.scopedId]?.closure.taskNodeMetadata?.cacheStatus ??
+    nodeExecutionsById?.[scopedId]?.closure?.taskNodeMetadata?.cacheStatus ??
     CatalogCacheStatus.CACHE_DISABLED;
 
   const dataProps = {
     nodeExecutionStatus,
     text: displayName,
     handles: [],
-    nodeType: node.type,
-    scopedId: node.scopedId,
+    nodeType,
+    scopedId,
     taskType,
     cacheStatus,
     onNodeSelectionChanged: () => {
       if (onNodeSelectionChanged) {
-        onNodeSelectionChanged([node.scopedId]);
+        onNodeSelectionChanged([scopedId]);
       }
     },
     onAddNestedView: () => {
       onAddNestedView({
-        parent: rootParentNode.scopedId,
-        view: node.scopedId,
+        parent: rootParentNode?.scopedId,
```

Comment:
`rootParentNode` is a `dNode` with `scopedId` being required field, adding `?` seems overly cautious to me 😄